e2e/tests/contracts.rs

```diff
@@ -1961,15 +1961,27 @@ async fn setup_node_with_high_price(historical_execution: bool) -> Result<Vec<Wa
         ..NodeConfig::default()
     };
 
+    let mut chain_config = ChainConfig {
+        consensus_parameters,
+        ..ChainConfig::default()
+    };
+
     if historical_execution {
-        node_config.database_type = DbType::RocksDb(None);
+        let temp_dir = tempfile::tempdir().expect("failed to make tempdir");
+        let temp_dir_name = temp_dir
+            .path()
+            .file_name()
+            .expect("failed to get file name")
+            .to_string_lossy()
+            .to_string();
+        let temp_database_path = temp_dir.path().join("db");
+
+        node_config.database_type = DbType::RocksDb(Some(temp_database_path));
         node_config.historical_execution = true;
+
+        chain_config.chain_name = temp_dir_name;
```

Comment:
Does a non-unique chain_name cause problems when running concurrent nodes?

I.e. maybe we should add a comment explaining why chain name needs to be set, how it is used, why does it need to be the name of a folder and whether it should be unique.

Also the tempdir will be deleted at the end of that scope. Maybe a comment explaining that we're aware of that and that we're using the tempdir to give us a guaranteed unique path for the db.

Reply:
Good catch. The unique name is not needed. https://github.com/FuelLabs/fuels-rs/pull/1628/commits/18767a678d2863d8b88bcfb1b28cbdb5ffb6ca74